### PR TITLE
🐛 Fixes for Branch Protection

### PR DESCRIPTION
**Pull request by azeemshaikh38**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Minor fixes for the `Branch-Protection` check.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No.

### CODE REVIEW

**Review by inferno-chromium — APPROVED** [review]

File: pkg/scorecard.go [hunk 1]
```diff
@@ -92,20 +92,16 @@ func RunScorecards(ctx context.Context,
...
-	if err != nil {
-		//nolint:wrapcheck
-		return ScorecardResult{}, err
-	}
-
-	if len(commits) == 0 {
-		//nolint:wrapcheck
-		return ScorecardResult{}, sce.Create(sce.ErrScorecardInternal, "no commits found")
+	var commitSHA string
+	if commits, err := repoClient.ListCommits(); err != nil && len(commits) > 0 {
```
Comment (laurentsimon) [inline pkg/scorecard.go:96]: I think this should be `err == nil`?

File: pkg/scorecard.go [hunk 2]
```diff
@@ -92,20 +92,16 @@ func RunScorecards(ctx context.Context,
...
+		commitSHA = commits[0].SHA
+	} else {
```
Comment (laurentsimon) [inline pkg/scorecard.go:98]: let's differentiate between the "no commit" error and the case `ListCommits()` returned an error?

File: checks/branch_protection.go [hunk 3]
```diff
@@ -84,40 +115,43 @@ func checkReleaseAndDevBranchProtection(ctx context.Context, r repositories, dl
...
-		if !p {
+		if !branch.GetProtected() {
 			protected = false
 			dl.Warn("branch protection not enabled for branch '%s'", b)
 		} else {
 			// The branch is protected. Check the protection.
 			score, err := getProtectionAndCheck(ctx, r, dl, ownerStr, repoStr, b)
 			if err != nil {
-				if errors.Is(err, errInternalBranchNotFound) {
+				if errors.As(err, &errInternalBranchNotFound) {
```
Comment (laurentsimon) [inline checks/branch_protection.go:154]: can you tell me the difference between `As` and `Is` and why `As` is better?

**Review by laurentsimon — APPROVED** [review]
In the PR, maybe add what `Minor fixes` mean. What is a bug in the code or some cleanup?

File: pkg/scorecard.go (hunk 2 above)
Comment (azeemshaikh38) [reply pkg/scorecard.go:98]: Done.

File: pkg/scorecard.go (hunk 1 above)
Comment (azeemshaikh38) [reply pkg/scorecard.go:96]: Done.

File: checks/branch_protection.go (hunk 3 above)
Comment (azeemshaikh38) [reply checks/branch_protection.go:154]: Ah you are right! I always considered `As` being the equivalent of `check error type only and not value`. Need to update all occurring of `errors.As` in codebase.

**Pull request merged**